storage-resize-images/functions/src/filters.ts

```diff
@@ -0,0 +1,56 @@
+import * as path from "path";
+
+import * as logs from "./logs";
+import config from "./config";
+import { supportedContentTypes } from "./resize-image";
+import { startsWithArray } from "./util";
+import { ObjectMetadata } from "firebase-functions/v1/storage";
+
+export function shouldResize(object: ObjectMetadata): boolean {
+  const { contentType } = object; // This is the image MIME type
+
+  const tmpFilePath = path.resolve("/", path.dirname(object.name)); // Absolute path to dirname
+
+  if (!contentType) {
+    logs.noContentType();
+    return false;
+  }
+
+  if (!contentType.startsWith("image/")) {
+    logs.contentTypeInvalid(contentType);
+    return false;
+  }
+
+  if (object.contentEncoding === "gzip") {
+    logs.gzipContentEncoding();
+    return false;
+  }
+
+  if (!supportedContentTypes.includes(contentType)) {
+    logs.unsupportedType(supportedContentTypes, contentType);
+    return false;
+  }
+
+  if (
+    config.includePathList &&
+    !startsWithArray(config.includePathList, tmpFilePath)
+  ) {
+    logs.imageOutsideOfPaths(config.includePathList, tmpFilePath);
+    return false;
+  }
+
+  if (
+    config.excludePathList &&
+    startsWithArray(config.excludePathList, tmpFilePath)
+  ) {
+    logs.imageInsideOfExcludedPaths(config.excludePathList, tmpFilePath);
+    return false;
+  }
+
+  if (object.metadata && object.metadata.resizedImage === "true") {
```

Comment:
So if you originally configure your images to be resized to 100x100 and backfill, and then reconfigure your images to be resized to 200x200, won't this return the wrong answer? Because, yes it was resized, but it's not the size that you just requested in the configuration. After fixing this, we should test that it works with config of a different size.

Reply:
I was concerned about this too, but i tested and it behaves appropriately. object.metadata.resizedImage is only set on the resized images (not on the original). On the next pass through, we'll see the original image without object.metadata.resizedImage, and resize it again.

The one case that is a bit wonky here is when 'delete original image' is set to true. In that case, the extension won't see any un-resized images & won't trigger correctly here. Need to think though a solution that handle this case.

Reply:
I think the correct solution is to document that only original images will be resized (i.e. we will not resize the already resized version of the original image). And add that if you choose to delete the originals after resizing, you would be required to upload them again if you decide you want a different size. Then they can make their choice between re-resizability and storage costs. Kevin will probably be helpful in how to phrase this well.

Reply:
I suggest adding this to the DELETE_ORIGINAL_FILE description: "If you enable this option, the extension won't be able to generate new resized images when you add image dimensions to your configuration."

or something like that

Reply:
Done, thanks for the guidance!